uniffi_bindgen/src/library_mode.rs

```diff
@@ -68,13 +68,27 @@ pub fn generate_external_bindings<T: BindingGenerator>(
     let cdylib_name = calc_cdylib_name(library_path);
     binding_generator.check_library_path(library_path, cdylib_name)?;
 
-    let mut sources = find_sources(
-        binding_generator,
-        &cargo_metadata,
-        library_path,
-        cdylib_name,
-        config_file_override,
-    )?;
+    let mut sources = find_components(&cargo_metadata, library_path)?
+        .into_iter()
+        .map(|(ci, package)| {
+            let crate_root = package
+                .manifest_path
+                .parent()
+                .context("manifest path has no parent")?;
+            let mut config = binding_generator
+                .new_config(&load_initial_config(crate_root, config_file_override)?)?;
+            if let Some(cdylib_name) = cdylib_name {
+                config.update_from_cdylib_name(cdylib_name);
+            }
+            config.update_from_ci(&ci);
+            Ok(Source {
+                config,
+                ci,
+                package,
+            })
+        })
+        .collect::<Result<Vec<_>>>()?;
```

Comment:
This step is also needed for non-external bindings, but I guess that's fine since `generate_bindings` just calls `generate_external_bindings` anyway.  What's stopping us from merging those 2 functions together?

Reply:
Nothing really - this is just a small step to make future steps easier to rationalize about.